### DISCUSSION TITLE
I have trained a model using my datasets, how can I use the model to detect anomaly in the video (.mp4) ?

### DISCUSSION THREAD

**Discussion by zhanghuayu-seu** [open-edge-platform/anomalib#1232] [Q&A] [answered]
### What is the motivation for this task?

Real-time anomaly detection

### Describe the solution you'd like

We use lightning_inference.py to achieve anomaly detection of a picture successfully, but find no proper functions to detect directly in the video.

**Comment by samet-akcay** [comment]
@djdameln, can you help here?

**Comment by alexriedel1** [comment, marked as answer]
You should write a function that reads single frames from the video and pass them to the inference function

**Comment by glucasol** [comment]
@zhanghuayu-seu Were You able to do that? How much FPS did you got with this real time inference?

**Comment by samet-akcay** [comment]
@glucasol, if speed is your concern for a video inference, you could potentially utilise OpenVINO's async inference, which significantly improves the FPS for such use cases. For more information, you could refer to this link:
https://docs.openvino.ai/2023.0/notebooks/115-async-api-with-output.html

Let us know if you have any more questions.
